config/s-nt.h

```diff
@@ -17,7 +17,7 @@
 
 #define OCAML_OS_TYPE "Win32"
 
-#ifdef __MINGW32__
+#if defined(__MINGW32__) || (defined(_MSC_VER) && _MSC_VER >= 1600)
```

Comment:
This can just be `#if defined(__MINGW32__) || _MSC_VER >= 1600` - even in MS world, undefined macros are expanded to `0` in comparisons.